ethers-providers/src/transports/common.rs

```diff
@@ -94,6 +94,33 @@ impl ResponseData<serde_json::Value> {
     }
 }
 
+/// Basic or bearer authentication in http or websocket transport
+///
+/// Use to inject username and password or an auth token into requests
+#[derive(Clone, Debug)]
+pub enum Authorization {
+    Basic(String),
+    Bearer(String),
+}
+
+impl Authorization {
+    pub fn basic(username: impl Into<String>, password: impl Into<String>) -> Self {
+        let auth_secret = base64::encode(username.into() + ":" + &password.into());
+        Self::Basic(auth_secret)
+    }
+
+    pub fn bearer(token: impl Into<String>) -> Self {
+        Self::Bearer(token.into())
+    }
+
+    pub(crate) fn into_auth_string(self) -> String {
+        match self {
+            Authorization::Basic(auth_secret) => format!("Basic {}", auth_secret),
+            Authorization::Bearer(token) => format!("Bearer {}", token),
+        }
+    }
```

Comment:
smol nit
this can be moved to a `impl Display for Authorization`

Reply:
I didn't do this on purpose, since I thought this is more of a transformation than just a way of displaying it. But I haven't a strong opinion on that. Happy to change it.